c/datasetlock.c

```diff
@@ -0,0 +1,129 @@
+
+
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+
+  SPDX-License-Identifier: EPL-2.0
+
+  Copyright Contributors to the Zowe Project.
+*/
+
+#ifndef __ZOWE_OS_WINDOWS
+#include <time.h>
+#endif
+
+#include "datasetlock.h"
+#include <sys/sem.h>
+#include "logging.h"
+
+void initLockResources() {
+    /* Create semaphore table for datasets */
+  for(int i=0; i < N_SEM_TABLE_ENTRIES; i++) {
+    sem_table_entry[i].sem_ID = 0;  /* initialise */
+  }
+  
+  for(int j=0; j < N_HBT_TABLE_ENTRIES; j++) {
+    hbt_table_entry[j].cnt = -1;  /* initialise */      
+  }
+
+  return;
+}
+
+int addUserToHbt(char user[8]){
```

Comment:
Should the heartbeat table also take a dataset as input rather than just a username? One scenario I pictured while testing was:
- A user has many datasets open in the editor but not all are being edited
- Because the datasets are open and with the heartbeat logic. all datasets the user has open in the editor will remain locked

Should the /datasetHeartbeat call take a dataset name as a parameter such that not all locks are refreshed? Perhaps the editor logic could be improved to only call datasetHeartbeat for the dataset currently being edited, or the most recently used datasets.

Reply:
yes, i think we should 
GET /datasetContent
and POST /datasetEnqueue - only when user intends to edit file.